exporter/zipkinexporter/zipkin.go

```diff
@@ -97,14 +94,27 @@ func zipkinEndpointFromNode(node *commonpb.Node, serviceName string, endpointTyp
 
 	var ip net.IP
 	ipv6Selected := false
-	if ipv4 := attributes[ipv4Key]; ipv4 != "" {
-		ip = net.ParseIP(ipv4)
-	} else if ipv6 := attributes[ipv6Key]; ipv6 != "" {
-		ip = net.ParseIP(ipv6)
-		ipv6Selected = true
+	if ipv4, ok := (*attributes)[ipv4Key]; ok {
+		if ipv4Str, ok := ipv4.(string); ok {
+			ip = net.ParseIP(ipv4Str)
+			delete(*attributes, ipv4Key)
```

Comment:
hum just realized that this is risky: in principle other exporters can try to use the attributes in parallel and deleting here becomes risky. The best approach perhaps is to exclude these attributes when preparing the zipkin span. I will look at this later.

Reply:
Yes. This is not in compliance with our data ownership model: https://github.com/open-telemetry/opentelemetry-collector/tree/master/exporter#data-ownership and will cause crashes or data corruption when multiple exporters are used.

Reply:
Fixed by avoiding the redundancy on the span generated by the exporter instead of changing the data. Eventually we will have to expand the data-ownership to cover exporters too.